File: app/src/main/java/com/example/lab7_recyclerview/MainActivity.java
package com.example.lab7_recyclerview;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private DatabaseHelper dbHelper;
    private StudentAdapter adapter;
    private RecyclerView recyclerView;
    private EditText etName;
    private EditText etEmail;
    private Button btnAdd;

    private Student selectedStudent = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dbHelper = new DatabaseHelper(this);

        etName = findViewById(R.id.etName);
        etEmail = findViewById(R.id.etEmail);
        btnAdd = findViewById(R.id.btnAdd);
        recyclerView = findViewById(R.id.recyclerView);

        setupRecyclerView();

        // Initial data if empty
        if (dbHelper.getAllStudents().isEmpty()) {
            for (int i = 1; i <= 10; i++) {
                dbHelper.addStudent(new Student("Student " + i, "student" + i + "@example.com"));
            }
            refreshData();
        }

        btnAdd.setOnClickListener(v -> {
            String name = etName.getText().toString();
            String email = etEmail.getText().toString();

            if (!name.isEmpty() && !email.isEmpty()) {
                if (selectedStudent == null) {
                    // Create
                    dbHelper.addStudent(new Student(name, email));
                    Toast.makeText(this, "Added successfully", Toast.LENGTH_SHORT).show();
                } else {
                    // Update
                    selectedStudent.setName(name);
                    selectedStudent.setEmail(email);
                    dbHelper.updateStudent(selectedStudent);
                    Toast.makeText(this, "Updated successfully", Toast.LENGTH_SHORT).show();
                    selectedStudent = null;
                    btnAdd.setText("Add Student");
                }
                clearInputs();
                refreshData();
            } else {
                Toast.makeText(this, "Please fill all fields", Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void setupRecyclerView() {
        adapter = new StudentAdapter(
                dbHelper.getAllStudents(),
                new StudentAdapter.OnStudentActionListener() {
                    @Override
                    public void onEditClick(Student student) {
                        selectedStudent = student;
                        etName.setText(student.getName());
                        etEmail.setText(student.getEmail());
                        btnAdd.setText("Update Student");
                    }

                    @Override
                    public void onDeleteClick(Student student) {
                        dbHelper.deleteStudent(student.getId());
                        refreshData();
                        Toast.makeText(MainActivity.this, "Deleted", Toast.LENGTH_SHORT).show();
                    }
                }
        );
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(adapter);
    }

    private void refreshData() {
        adapter.updateData(dbHelper.getAllStudents());
    }

    private void clearInputs() {
        etName.getText().clear();
        etEmail.getText().clear();
    }
}

File: app/src/main/java/com/example/lab7_recyclerview/MainActivity.kt
package com.example.lab7_recyclerview

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private lateinit var dbHelper: DatabaseHelper
    private lateinit var adapter: StudentAdapter
    private lateinit var recyclerView: RecyclerView
    private lateinit var etName: EditText
    private lateinit var etEmail: EditText
    private lateinit var btnAdd: Button
    
    private var selectedStudent: Student? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        dbHelper = DatabaseHelper(this)
        
        etName = findViewById(R.id.etName)
        etEmail = findViewById(R.id.etEmail)
        btnAdd = findViewById(R.id.btnAdd)
        recyclerView = findViewById(R.id.recyclerView)

        setupRecyclerView()
        
        // Initial data if empty
        if (dbHelper.getAllStudents().isEmpty()) {
            for (i in 1..10) {
                dbHelper.addStudent(Student(name = "Student $i", email = "student$i@example.com"))
            }
            refreshData()
        }

        btnAdd.setOnClickListener {
            val name = etName.text.toString()
            val email = etEmail.text.toString()

            if (name.isNotEmpty() && email.isNotEmpty()) {
                if (selectedStudent == null) {
                    // Create
                    dbHelper.addStudent(Student(name = name, email = email))
                    Toast.makeText(this, "Added successfully", Toast.LENGTH_SHORT).show()
                } else {
                    // Update
                    val studentToUpdate = selectedStudent!!
                    studentToUpdate.name = name
                    studentToUpdate.email = email
                    dbHelper.updateStudent(studentToUpdate)
                    Toast.makeText(this, "Updated successfully", Toast.LENGTH_SHORT).show()
                    selectedStudent = null
                    btnAdd.text = "Add Student"
                }
                clearInputs()
                refreshData()
            } else {
                Toast.makeText(this, "Please fill all fields", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun setupRecyclerView() {
        adapter = StudentAdapter(
            dbHelper.getAllStudents(),
            onEditClick = { student ->
                selectedStudent = student
                etName.setText(student.name)
                etEmail.setText(student.email)
                btnAdd.text = "Update Student"
            },
            onDeleteClick = { student ->
                dbHelper.deleteStudent(student.id)
                refreshData()
                Toast.makeText(this, "Deleted", Toast.LENGTH_SHORT).show()
            }
        )
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = adapter
    }

    private fun refreshData() {
        adapter.updateData(dbHelper.getAllStudents())
    }

    private fun clearInputs() {
        etName.text.clear()
        etEmail.text.clear()
    }
}
